src/frame_buffer.rs

```diff
@@ -67,6 +74,18 @@ impl<'a> FrameBuffer<'a> {
         self.dimensions
     }
 
+    /// Return the origin of the frame buffer.
+    pub fn origin(&self) -> (i32, i32) {
```

Comment:
It's not clear which origin this is, what it's relative to, or why the framebuffer needs to be aware of it. Upstream does not seem to have a similar member or accessor.

Reply:
This origin is needed to calculate the offset to the data pointer in `insert_raw`.

Reply:
The documentation, and indeed naming, should make it clear what this value is.

Reply:
Sounds good, what do you suggest?

Reply:
As an alternative, the `origin` field (or the derived `offset`) may be added as a parameter in all the `insert_channel*` calls.

Would it be more acceptable to you? I still prefer the option of it being a member of the framebuffer struct as it is easier to validate and less prone to errors.

Reply:
I don't necessarily object to the approach taken here, I just haven't fully grasped how it works yet. I'm totally on board with papering over ill-conceived upstream APIs when it's feasible to do so and doesn't make things even more confusing elsewhere; we do after all already do similar-looking things with the `dimensions` field.